File: backend/src/main/java/br/com/ficaespertoapp/backend/infrastructure/persistence/entity/Post.java
package br.com.ficaespertoapp.backend.infrastructure.persistence.entity;

import br.com.ficaespertoapp.backend.domain.enums.PostCategory;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "posts")
@Getter @Setter
@NoArgsConstructor  @AllArgsConstructor
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private String author;

    @Column(nullable = false)
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PostCategory category;
}

File: frontend/mobile/src/models/post.ts
export class Post {
  id?: number;
  title?: string;
  description?: string;
  content?: string;
  author?: string;
  imageUrl?: string;
  category?: string;

  constructor(init?: Partial<Post>) {
    Object.assign(this, init);
  }

  static fromJson(json: any): Post {
    return new Post({
      id: json.id,
      title: json.title,
      description: json.description,
      content: json.content,
      author: json.author,
      imageUrl: json.imageUrl,
      category: json.category,
    });
  }

  toJson(): any {
    return {
      id: this.id,
      title: this.title,
      description: this.description,
      content: this.content,
      author: this.author,
      imageUrl: this.imageUrl,
      category: this.category,
    };
  }
}
